src/core/search/indices.cc

```diff
@@ -0,0 +1,69 @@
+// Copyright 2023, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#include "core/search/indices.h"
+
+#include <absl/container/flat_hash_set.h>
+#include <absl/strings/ascii.h>
+#include <absl/strings/numbers.h>
+
+#include <algorithm>
+#include <regex>
+
+#include "base/logging.h"
+
+namespace dfly::search {
+
+using namespace std;
+
+namespace {
+
+// Get all words from text as matched by regex word boundaries
+vector<string> GetWords(string_view text) {
+  std::regex rx{"\\b.*?\\b", std::regex_constants::icase};
+  std::cregex_iterator begin{text.data(), text.data() + text.size(), rx}, end{};
+
+  absl::flat_hash_set<string> words;
```

Comment:
I want to track the words in lowercase, so I can remove duplicates instantly. How can I track them without conversions? Store the temporary lowercase in the buffer and binary search if the index already contains the doc-id? Maybe

Reply:
Either way we will resort to filtering duplicates ahead because we will need to count them properly and maybe fetch some auxiliary info (context, etc)

Reply:
I missed this point. Maybe worth adding the comment here why we create our own strings.